src/Integrations/Integrations/Magento/MagentoIntegration.php

```diff
@@ -0,0 +1,704 @@
+                    $span->meta['magento.block.area'] = $block->getArea();
+                }
+
+                $moduleName = empty($moduleName) ? '<core>' : $moduleName;
+
+                $blockName = $block->getNameInLayout();
+                $span->resource = "{$moduleName}:{$blockName}";
+
+                $span->meta['magento.block.module'] = $moduleName;
+                $span->meta['magento.block.name'] = $blockName;
+
+                $cacheLifetime = $block->getCacheLifetime();
+                if ($cacheLifetime !== null) {
+                    $span->meta['magento.block.cache_lifetime'] = $cacheLifetime;
+                }
+
+                $class = MagentoIntegration::getRealClass($block);
+                $span->meta['magento.block.class'] = $class;
+
+                // For Legacy, see Magento\Widget\Model\Widget\Instance::generateLayoutUpdateXml
+                if (strlen($blockName) === 32
+                    && $class === 'Magento\Cms\Block\Widget\Block'
+                    && $integration->calculateEntropy($blockName) > 4.0) {
+                    $span->resource = "$moduleName:<widget>";
+                }
+            }
+        );
+
+        // Response
+        trace_method(
+            'Magento\Framework\App\ResponseInterface',
+            'sendResponse',
+            function (SpanData $span) {
+                MagentoIntegration::setCommonSpanInfo($span, 'magento.response.send');
+            }
+        );
+
+        // Exception handling
+        $integration = $this;
+        hook_method(
+            'Magento\Framework\AppInterface',
+            'catchException',
+            null,
+            function ($http, $scope, $args) use ($integration) {
+                $rootSpan = root_span();
+                if ($rootSpan !== null) {
+                    $integration->setError($rootSpan, $args[1]);
+                }
+
+            }
+        );
+
+        // Identify current user (userid, email)
+        hook_method(
+            'Magento\Customer\Model\Session',
+            'isLoggedIn',
+            null,
+            function ($session, $scope, $args, $isLoggedIn) {
+                if ($isLoggedIn && root_span() !== null) {
+                    /** @var Magento\Customer\Model\Data\Customer $customer */
+                    $customer = $session->getCustomer();
+                    $userId = $customer->getId();
+                    $email = $customer->getEmail();
+                    set_user($userId, empty($email) ? null : ['email' => $email]);
+                }
+            }
+        );
+
+
+        // Events
+        trace_method(
+            'Magento\Framework\Event\ObserverInterface',
+            'execute',
+            function (SpanData $span) {
+                $class = get_class($this);
+                if ($class !== 'Magento\PageCache\Observer\ProcessLayoutRenderElement') {
+                    MagentoIntegration::setCommonSpanInfo($span, 'magento.event.execute', $class);
```

Comment:
Can we hook `Magento\Framework\View\Element\UiComponent\SubjectInterface::notify` to add the name of the event? Might need to be implemented as a stack - push on begin, pop on end - and this hook here then peeks at the top of the notify-stack.